### ISSUE TITLE
[super_clipboard] Can't build as a flutter module

### ISSUE THREAD

**Issue by jifang** [superlistapp/super_native_extensions#379]
When build the super_clipboard as a flutter module and integrate into the existing android project, I am getting the following error

```
FAILURE: Build failed with an exception.

* Where:
Script '.../irondash_engine_context-0.5.3/cargokit/gradle/plugin.gradle' line: 114

* What went wrong:
A problem occurred evaluating script.
> Failed to apply plugin class 'CargoKitPlugin'.
   > Could not get unknown property 'applicationVariants' for extension 'android' of type com.android.build.gradle.LibraryExtension.
```

Not only the irondash library, the `super_native_extensions-0.8.16/cargokit/gradle/plugin.gradle` has the same problem.

It looks like that the gradle plugin assumes that it must be part of an application target. This issue can be fixed by modifying the gradle script like following

```
        def isApplication = project.plugins.hasPlugin('com.android.application')
        def variants = isApplication ? project.android.applicationVariants : project.android.libraryVariants

        variants.all { variant ->
                ... ...
         }
```

Is there a non intrusive way to solve this issue?

**Comment by knopp** [comment]
If you can make a PR for [cargokit](https://github.com/irondash/cargokit) that would be very welcome. Otherwise this will be resolved once things get migrated to native assets (currently experimental in Flutter). That way cargokit will no longer be required.

**Comment by tibo-lumapps** [comment]
Hello, i have the same issue.
I saw you made a PR, did you find any temporary fix for now ?

**Comment by jifang** [comment]
My temporary solution is to change the source in the pub cache folder lol. Sent from my iPhoneOn May 31, 2024, at 19:07, tibo-lumapps ***@***.***> wrote:﻿
Hello, i have the same issue.
I saw you made a PR, did you find any temporary fix for now ?

**Comment by jifang** [comment]
> If you can make a PR for [cargokit](https://github.com/irondash/cargokit) that would be very welcome. Otherwise this will be resolved once things get migrated to native assets (currently experimental in Flutter). That way cargokit will no longer be required.

@knopp I've already submitted the PR. Would you mind to take a look? It's a minor change and should be pretty safe.

**Comment by knopp** [comment]
@jifang, thank you. I merged it in cargokit, I'll try to release updated irondash_engine_context and super_native_extensions today.

**Comment by knopp** [comment]
Should be fixed in 0.8.17.
